rest-service/manager_rest/rest/resources_v3_1/searches.py

```diff
@@ -0,0 +1,84 @@
+from flask import request
+
+from manager_rest.security import SecuredResource
+from manager_rest.security.authorization import authorize
+from manager_rest.storage import models, get_storage_manager
+
+from ..responses_v2 import ListResponse
+from .. import rest_decorators, rest_utils
+from ..filters_utils import (create_filter_rules_list,
+                             get_filter_rules_from_filter_id)
+
+
+class ResourceSearches(SecuredResource):
+    def post(self, resource_model, _include, filters, pagination, sort,
+             all_tenants, search, filter_id, **kwargs):
+        """List resource items"""
+        get_all_results = rest_utils.verify_and_convert_bool(
+            '_get_all_results',
+            request.args.get('_get_all_results', False)
+        )
+        request_schema = {'filter_rules': {'optional': False, 'type': list}}
+        request_dict = rest_utils.get_json_and_verify_params(request_schema)
+
+        filter_rules = _get_filter_rules(request_dict['filter_rules'],
+                                         resource_model, filter_id)
+
+        result = get_storage_manager().list(
+            resource_model,
+            include=_include,
+            filters=filters,
+            substr_filters=search,
+            pagination=pagination,
+            sort=sort,
+            all_tenants=all_tenants,
+            get_all_results=get_all_results,
+            filter_rules=filter_rules
+        )
+
+        return ListResponse(items=result.items, metadata=result.metadata)
+
+
+def _get_filter_rules(raw_filter_rules, resource_model, filter_id):
```

Comment:
This function, as `create_filter_rules_list`, doesn't validate the uniqueness of filter rules. I.e. the user **can** provide the same filter rule in the list (and I even created a test for it). For some reason, I'm still not convinced this is the right behavior and we might want to limit it. On the rest service side, I don't think having duplicated filter rules would cause an issue as this only means duplicated conditions in the query. @tehasdf what do you think?

Reply:
I don't really see a reason to disallow it, we might deduplicate it

Reply:
I don't see a reason to limit it on the user side, but we can drop duplicates from the query before we send it

Reply:
@tehasdf, @glukhman, thanks. Do you see any straightforward way of doing it? I came up with a method which you can see [here](https://github.com/cloudify-cosmo/cloudify-manager/pull/2748/commits/569ed810575b3275269a1dc0a025bf1300b81036#diff-90f5a7c8fde21a321436b1de427fe0c5ed49cbbddee075345edb57fa681599c2R129) and [here](https://github.com/cloudify-cosmo/cloudify-manager/pull/2748/commits/569ed810575b3275269a1dc0a025bf1300b81036#diff-53b96b3c8ef18ae614ee5f6489124011cab623cdb669ca26066e2b48dcc288aeR47), but it's not the most efficient one.

Reply:
it'd have to go through a set, yeah. If we have a way to make them tuples..

Reply:
Updated the FilterRule object to be hashable.